### PR TITLE
Change regex capture variable from \n to $n

### PR DESCRIPTION
**Pull request by mikeschinkel**
When inspecting code with PhpStorm it flags the code as an error when the \n stype of capture variable is used, and in at least the news2html file I am pretty sure that code was broken.  This PR changes those capture variables to $n format.

Ref: @cmb69's [comment](https://github.com/php/web-php/pull/416#issuecomment-883555514) on other PR.

### CODE REVIEW

File: bin/news2html [hunk 1]
```diff
@@ -70,7 +70,7 @@ foreach($entries as $module => $items) {
 		// convert bug numbers
 		$item = preg_replace(
 			array('/Fixed bug #([0-9]+)/', '/Fixed PECL bug #([0-9]+)/', '/Implemented FR #([0-9]+)/', '/GitHub PR #([0-9]+)/'),
-			array('<?php bugfix(\1); ?>', '<?php peclbugfix(\1); ?>', '<?php implemented(\1); ?>', '<?php githubissuel(\'php/php-src\',\1); ?>'),
+			array('<?php bugfix("$1"); ?>', '<?php peclbugfix("$1"); ?>', '<?php implemented("$1"); ?>', '<?php githubissuel(\'php/php-src\',"$1"); ?>'),
```
[the remark this answers is not in the document]
Comment (mikeschinkel) [reply bin/news2html:73]: My motivation for that PR as well as most of the rest of the PRs I submitted today was to eliminate errors or warnings generated when static analysis, which I did by running code inspection via PhpStorm.

In the case of line `73` it was claiming an error and the way that I chose to get around the error was to put in double quotes because otherwise is it not valid syntax.  

Notice that the `bugfix()` call comes after a `<?php` so AFAICT without know the context this is run in is that it needs to be valid PHP syntax, and neither `foo(\1)` nor `foo($1)` are valid PHP syntax, but `foo("$1")` is.

Also, the functions take the parameter value and concatenate into a URL, so it makes sense to pass in a string.

That said, I wondered why we needed to generate code that would call `bugfix()` at runtime to generate the links and why it wouldn't be better just to generate static HTML links. The code would certainly be simplistic. 

One reason might be the developer viewed that it would be easier to fix the functions if this URLs changed, but I think it would be just as easy to do a regex search/replace on the files in the repo.  

Still, I did not want to make that kind of change as my motivation for this round of changes was to just get rid of all the warnings when doing code inspections, and obviously to fix any errors that could be fixed.

File: archive/2015.php [hunk 2]
```diff
@@ -20,5 +20,5 @@
 
 print_news($NEWS_ENTRIES, array("conferences", "cfp", "frontpage", "nofrontpage"), 500, 2015);
 
-/* %s/<a href="\(.*\)"><img src="\/images\/news\/\(.*\)" alt="\(.*\)" width.*><\/a>/<?php news_image("\1", "\2", "\3"); ?>/g */
+/* %s/<a href="\(.*\)"><img src="\/images\/news\/\(.*\)" alt="\(.*\)" width.*><\/a>/<?php news_image('$1', '$2', '$3'); ?>/g */
```
[the remark this answers is not in the document]
Comment (mikeschinkel) [reply archive/2015.php:23]: @salathe — So:

1.) Are those instructions still needed, or has the archiving been done?
2.) If not, maybe we should delete these comment lines?
3.) If still needed, do my changes break the code?

File: bin/news2html (hunk 1 above)
[the remark this answers is not in the document]
Comment (salathe) [reply bin/news2html:73]: Using `\1` was valid syntax. Using `$1` is valid syntax.  Note that there are *replacement strings* for [`preg_replace()`](https://php.net/preg_replace).  The `\1` or `$1` placeholders will get the actual values from the regex match substituted.

If the input is `Fixed bug #123` then the output (using `$1` without added double-quotes) is `<?php bugfix(123); ?>`.  With your change as-is, the output is `<?php bugfix("123"); ?>`.  While there is nothing, as far as I understand, *wrong* with that output, it is a change... so I wanted to question it.

In summary: I'm fine with changing `\1` to `$1`, since the latter is the *preferred* replacement placeholder style (see [`preg_replace()`](https://php.net/preg_replace)).  I don't see the point of wrapping the value in quotes.

File: archive/2015.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (salathe) [reply archive/2015.php:23]: 1) Yes, they are still needed as a reminder for whoever is the next person to archive that year.
2) In already-archived years that have the comment (only `2014.php`), feel free to delete the comment line.
3) Yes, your changes break the "code".

File: bin/news2html (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikeschinkel) [reply bin/news2html:73]: @salathe — These screenshots show why I changed it.  PhpStorm's static analysis evidently gets confused by the syntax, and as such I got confused by the syntax. 

Cursor_and_web-php_–_news2html

web-php_–_news2html

So the point in wrapping in quotes is so that static analysis doesn't accidentally flag it as an error and have the next person duplicate this discussion with someone else in the future. Given that fact that all the called functions are doing is string concatenation it doesn't seem like the fact I changed to a string will have any chance of breaking anything since the functions do not have type hints.

web-php_–_changelogs_inc

That said, I just devoted about 12 hours to working on these PRs and am exhausted, so the last thing I want to do is bikeshed over something like this. If it is important to change it back to the way it was before where PhpStorm still flags it as an error when doing code inspection then I'll just change it back and move on.

File: archive/2015.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikeschinkel) [reply archive/2015.php]: @salathe — I have reverted those changes and deleted the comment line from `2014.php`.

File: include/layout.inc [hunk 3]
```diff
@@ -51,7 +51,7 @@ function highlight_php_trimmed($code, $return = false)
 {
     $code = "<?php\n" . $code;
     $highlighted_code = highlight_php($code, true);
-    $highlighted_code = preg_replace("/\&lt;\?php(\<br \/\>)+/", '', $highlighted_code, 1);
+    $highlighted_code = preg_replace("/&lt;\?php(<br \/>)+/", '', $highlighted_code, 1);
```
Comment (mikeschinkel) [inline include/layout.inc:54]: In case anyone want to understand this, PhpStorm flagged these backslashes as unnecessary.

File: bin/news2html (hunk 1 above)
[the remark this answers is not in the document]
Comment (salathe) [reply bin/news2html:73]: Thanks for the explanation. In general, I prefer not to include code changes *just* to work around a bug in an IDE. In this case in particular, thankfully, the workaround is harmless.

[the remark this answers is not in the document]
Comment (mikeschinkel) [reply bin/news2html:73]: @salathe — Fair point.  

In this case ironically we might consider the "workaround" actually makes this code more correct, since it really is a string that it is accepting.

That said, I want to revisit the question, should this be generating code that calls those functions at page load type dynamically, or shouldn't we just go ahead and generate the links and URLs statically?  That would make the code a lot clearer and cleaner, too?

_(Which brings up the bigger question; is the team currently happy with the existing website and codebase, or would the team see benefit in modernizing the code and adding functionality to the the site? )_

[the remark this answers is not in the document]
Comment (cmb69) [reply bin/news2html:73]: I'm pretty sure that the static analyzer is picky about the single backslash inside single-quotes.
